recipes/sail/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+from conans.errors import ConanInvalidConfiguration
+from conan.tools.files import rename
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.43.0"
+
+class SAILConan(ConanFile):
+    name = "sail"
+    version = "0.9.0"
+    description = "The missing small and fast image decoding library for humans (not for machines)"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://sail.software"
+    topics = ( "image", "encoding", "decoding", "graphics" )
+    license = "MIT"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "thread_safe": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "thread_safe": True,
+    }
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    exports_sources = "CMakeLists.txt"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("giflib/5.2.1")
+        self.requires("jasper/2.0.33")
+        self.requires("libjpeg/9d")
+        self.requires("libpng/1.6.37")
+        self.requires("libtiff/4.3.0")
+        self.requires("libwebp/1.2.2")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  strip_root=True, destination=self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = "ON" if self.options.get_safe("fPIC", default=True) or self.options.shared else "OFF"
+        self._cmake.definitions["SAIL_BUILD_APPS"]     = "OFF"
+        self._cmake.definitions["SAIL_BUILD_EXAMPLES"] = "OFF"
+        self._cmake.definitions["SAIL_BUILD_TESTS"]    = "OFF"
+        self._cmake.definitions["SAIL_COMBINE_CODECS"] = "ON"
+        self._cmake.definitions["SAIL_THREAD_SAFE"]    = "ON" if self.options.thread_safe else "OFF"
```

Comment:
```suggestion
        self._cmake.definitions["CMAKE_POSITION_INDEPENDENT_CODE"] = self.options.get_safe("fPIC", True)
        self._cmake.definitions["SAIL_BUILD_APPS"]     = False
        self._cmake.definitions["SAIL_BUILD_EXAMPLES"] = False
        self._cmake.definitions["SAIL_BUILD_TESTS"]    = False
        self._cmake.definitions["SAIL_COMBINE_CODECS"] = False
        self._cmake.definitions["SAIL_THREAD_SAFE"]    = self.options.thread_safe
```

Reply:
There is a static or object lib (sail-codecs) linked into a shared lib when `BUILD_SHARED_LIBS ON`? Therefore, it should be fixed in CMakeLists (set `POSITION_INDEPENDENT_CODE` property to ON for this static or object target when `BUILD_SHARED_LIBS ON`).

Reply:
Thanks for the update! I'll check PIC in my code.

Could you please also check why it cannot be built against `libavif`? The log file: https://c3i.jfrog.io/c3i/misc/logs/pr/10364/48-linux-gcc/sail/0.9.0//cb507183b78f80a724b865c573bc22e9e5d25bc2-build.txt Is it a problem in my recipe?

Reply:
I guess because https://github.com/HappySeaFox/sail/blob/974c53ff65c7e5958c232b762353122e20db5fbf/src/sail-codecs/avif/avif.cmake is fragile, and actually broken when libavif is static since there are transitive dependencies.

libavif provides a CMake config file, you should rely on it:

```cmake
find_package(libavif REQUIRED CONFIG)
target_link_libraries(foo PRIVATE avif)
```

Reply:
Thank you so much, @SpaceIm !